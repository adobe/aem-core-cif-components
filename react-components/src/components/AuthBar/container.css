.enter {
    transition-duration: 384ms;
    transition-timing-function: var(--venia-anim-in);
}

.exit {
    transition-duration: 192ms;
    transition-timing-function: var(--venia-anim-out);
}

.hidden {
    opacity: 0;
    visibility: hidden;
}
.visible {
    opacity: 1;
    visibility: visible;
}

.modal {
    composes: exit hidden;
    background-color: white;
    bottom: 0;
    left: 0;
    overflow: auto;
    position: absolute;
    right: 0;
    top: 3.5rem;
    transform: translate3d(-100%, 0, 0);
    transition-property: opacity, transform, visibility;
}

.modal_open {
    composes: modal enter visible;
    transform: translate3d(0, 0, 0);
}
