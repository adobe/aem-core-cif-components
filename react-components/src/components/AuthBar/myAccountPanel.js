/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';

import SignIn from '../SignIn';
import MyAccount from '../MyAccount';
import ForgotPassword from '../ForgotPassword';
import CreateAccount, { CreateAccountSuccess } from '../CreateAccount';
import ChangePassword from '../ChangePassword';

import classes from './myAccountPanel.css';
import { string, shape, func } from 'prop-types';

const AuthModal = ({ view = 'MENU', api, children }) => {
    const {
        showSignIn,
        showMyAccount,
        showMenu,
        showForgotPassword,
        showChangePassword,
        showCreateAccount,
        showAccountCreated,
        handleBack
    } = api;

    let child;

    switch (view) {
        case 'SIGN_IN':
            child = (
                <SignIn
                    showMyAccount={showMyAccount}
                    showForgotPassword={showForgotPassword}
                    showCreateAccount={showCreateAccount}
                />
            );
            break;
        case 'MY_ACCOUNT':
            child = (
                <MyAccount
                    showMenu={showMenu}
                    showChangePassword={showChangePassword}
                    showAccountInformation={() => {}}>
                    {children}
                </MyAccount>
            );
            break;
        case 'CHANGE_PASSWORD':
            child = <ChangePassword showMyAccount={showMyAccount} handleCancel={handleBack} />;
            break;
        case 'FORGOT_PASSWORD':
            child = <ForgotPassword onClose={showMenu} onCancel={handleBack} />;
            break;
        case 'CREATE_ACCOUNT':
            child = (
                <CreateAccount
                    showMyAccount={showMyAccount}
                    showAccountCreated={showAccountCreated}
                    handleCancel={handleBack}
                />
            );
            break;
        case 'ACCOUNT_CREATED':
            child = <CreateAccountSuccess showSignIn={showSignIn} />;
    }

    return <div className={classes.root}>{child}</div>;
};

AuthModal.propTypes = {
    view: string,
    api: shape({
        showSignIn: func,
        showMyAccount: func,
        showMenu: func,
        showForgotPassword: func,
        showChangePassword: func,
        showCreateAccount: func,
        showAccountCreated: func,
        handleBack: func
    })
};

export default AuthModal;
