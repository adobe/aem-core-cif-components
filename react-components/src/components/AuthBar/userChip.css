/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 ~ Copyright 2021 Adobe
 ~
 ~ Licensed under the Apache License, Version 2.0 (the "License");
 ~ you may not use this file except in compliance with the License.
 ~ You may obtain a copy of the License at
 ~
 ~     http://www.apache.org/licenses/LICENSE-2.0
 ~
 ~ Unless required by applicable law or agreed to in writing, software
 ~ distributed under the License is distributed on an "AS IS" BASIS,
 ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 ~ See the License for the specific language governing permissions and
 ~ limitations under the License.
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
.root {
    width: 100%;
}

.content {
    align-items: center;
    display: grid;
    font-size: 0.875rem;
    height: 5.5rem;
    gap: 0.5rem;
    grid-auto-flow: column;
    grid-template-columns: 3rem 1fr 3rem;
    line-height: 1rem;
    padding: 0 0.5rem;
    text-align: center;
}

.avatar {
}

.user {
    display: grid;
    gap: 0.25rem;
    grid-auto-flow: row;
    text-align: left;
}

.fullName {
}

.email {
    color: rgb(var(--venia-text-alt));
}

.icon {
}
