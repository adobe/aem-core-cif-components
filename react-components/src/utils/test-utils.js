/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import { render } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { I18nextProvider } from 'react-i18next';

import i18n from '../../__mocks__/i18nForTests';
import ConfigContextProvider from '../context/ConfigContext';
import UserContextProvider from '../context/UserContext';

import mutationChangePassword from './mocks/mutationChangePassword';
import mutationCreateCustomer from './mocks/mutationCreateCustomer';
import mutationCreateDuplicateCustomer from './mocks/mutationCreateCustomerDuplicate';
import mutationDeleteCustomerAddress from './mocks/mutationDeleteCustomerAddress';
import mutationGenerateToken from './mocks/mutationGenerateToken';
import mutationGenerateTokenWrongPassword from './mocks/mutationGenerateTokenWrongPassword';
import mutationMergeCarts from './mocks/mutationMergeCarts';
import mutationPlaceOrder from './mocks/mutationPlaceOrder';
import mutationRevokeToken from './mocks/mutationRevokeToken';
import mutationShippingAddress from './mocks/mutationShippingAddress';
import queryCart from './mocks/queryCart';
import queryCountries from './mocks/queryCountries';
import queryCustomerCart from './mocks/queryCustomerCart';
import queryCustomerDetails from './mocks/queryCustomerDetails';
import queryCustomerInformation from './mocks/queryCustomerInformation';
import queryEmptyCart from './mocks/queryEmptyCart';
import queryNewCart from './mocks/queryNewCart';

const defaultMocks = [
    mutationChangePassword,
    mutationCreateCustomer,
    mutationCreateDuplicateCustomer,
    mutationDeleteCustomerAddress,
    mutationGenerateToken,
    mutationGenerateTokenWrongPassword,
    mutationMergeCarts,
    mutationPlaceOrder,
    mutationRevokeToken,
    mutationShippingAddress,
    queryCart,
    queryCountries,
    queryCustomerCart,
    queryCustomerDetails,
    queryCustomerInformation,
    queryEmptyCart,
    queryNewCart
];

const defaultConfig = {
    storeView: 'default',
    graphqlEndpoint: 'none'
};

// eslint-disable-next-line react/display-name
const allProviders = (config, userContext, mocks) => ({ children }) => {
    return (
        <MockedProvider mocks={mocks} addTypename={false}>
            <ConfigContextProvider config={config || defaultConfig}>
                <UserContextProvider initialState={userContext}>
                    <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
                </UserContextProvider>
            </ConfigContextProvider>
        </MockedProvider>
    );
};

/* Wrap all the React components tested with the library in a mocked Apollo provider */
const customRender = (ui, options = {}) => {
    const { config, userContext, mocks = defaultMocks, ...renderOptions } = options;
    return render(ui, { wrapper: allProviders(config, userContext, mocks), ...renderOptions });
};

export * from '@testing-library/react';
export { customRender as render };
