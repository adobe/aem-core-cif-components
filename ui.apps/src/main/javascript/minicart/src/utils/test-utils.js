/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/

import React from 'react';
import { render } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';

import QUERY_CART_DETAILS from '../queries/query_cart_details.graphql';

const emptyCartId = 'empty';
const mockCartId = '123ABC';

const mocks = [
    {
        request: {
            query: QUERY_CART_DETAILS,
            variables: {
                cartId: emptyCartId
            }
        },
        result: {
            data: {
                cart: {
                    email: null,
                    shipping_addresses: [],
                    prices: {
                        grand_total: {
                            currency: 'USD',
                            value: 0
                        }
                    },
                    selected_payment_method: {
                        code: '',
                        title: ''
                    },
                    billing_address: {
                        city: null,
                        country: {
                            code: null
                        },
                        lastname: null,
                        firstname: null,
                        region: {
                            code: null
                        },
                        street: [''],
                        postcode: null,
                        telephone: null
                    },
                    available_payment_methods: [
                        {
                            code: 'cashondelivery',
                            title: 'Cash On Delivery'
                        },
                        {
                            code: 'banktransfer',
                            title: 'Bank Transfer Payment'
                        },
                        {
                            code: 'checkmo',
                            title: 'Check / Money order'
                        },
                        {
                            code: 'free',
                            title: 'No Payment Information Required'
                        }
                    ],
                    items: []
                }
            }
        }
    },
    {
        request: {
            query: QUERY_CART_DETAILS,
            variables: {
                cartId: mockCartId
            }
        },
        result: {
            data: {
                prices: {
                    grand_total: {
                        currency: 'USD',
                        value: 0
                    }
                },
                items: []
            }
        }
    }
];

const AllProviders = ({ children }) => {
    return (
        <MockedProvider mocks={mocks} addTypename={false}>
            {children}
        </MockedProvider>
    );
};

/* Wrap all the React components tested with the library in a mocked Apollo provider */
const customRender = (ui, options) => render(ui, { wrapper: AllProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
