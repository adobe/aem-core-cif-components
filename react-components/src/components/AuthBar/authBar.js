/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React, { useEffect } from 'react';
import { useTranslation } from 'react-i18next';

import Button from '../Button';
import classes from './authBar.css';
import { useUserContext } from '../../context/UserContext';
import UserChip from './userChip';
import { func } from 'prop-types';
import * as dataLayerUtils from '../../utils/dataLayerUtils';

const AuthBar = ({ showMyAccount, showSignIn }) => {
    const [{ currentUser, isSignedIn }, { getUserDetails }] = useUserContext();

    useEffect(() => {
        if (isSignedIn && currentUser.email === '') {
            getUserDetails();
        }
    }, [getUserDetails]);

    useEffect(() => {
        if (!isSignedIn) {
            dataLayerUtils.pushData({ user: null });
        } else if (isSignedIn && currentUser.email !== '') {
            dataLayerUtils.pushData({ user: currentUser });
        }
    }, [isSignedIn, currentUser]);

    const [t] = useTranslation('account');

    const disabled = false;

    const content = isSignedIn ? (
        <UserChip currentUser={currentUser} showMyAccount={showMyAccount} />
    ) : (
        <Button disabled={!!disabled} priority="high" onClick={showSignIn}>
            {t('account:sign-in', 'Sign In')}
        </Button>
    );
    return <div className={classes.root}>{content}</div>;
};

AuthBar.propTypes = {
    showMyAccount: func,
    showSignIn: func
};

export default AuthBar;
