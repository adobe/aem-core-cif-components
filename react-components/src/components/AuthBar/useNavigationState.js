/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import { useState, useCallback } from 'react';
import { useTranslation } from 'react-i18next';
import { useEventListener } from '../../utils/hooks';
import { useConfigContext } from '../../context/ConfigContext';

// DOM events that are used to talk to the navigation panel
const events = {
    START_ACC_MANAGEMENT: 'aem.accmg.start',
    EXIT_ACC_MANAGEMENT: 'aem.accmg.exit'
};

const ancestors = {
    CREATE_ACCOUNT: 'SIGN_IN',
    FORGOT_PASSWORD: 'SIGN_IN',
    ACCOUNT_CREATED: 'MENU',
    MY_ACCOUNT: 'MENU',
    CHANGE_PASSWORD: 'MY_ACCOUNT',
    SIGN_IN: 'MENU',
    MENU: null
};

const stepTitles = {
    CREATE_ACCOUNT: t => t('account:create-account', 'Create account'),
    FORGOT_PASSWORD: t => t('account:password-recovery', 'Password recovery'),
    CHANGE_PASSWORD: t => t('account:change-password', 'Change Password'),
    MY_ACCOUNT: t => t('account:my-account', 'My account'),
    ACCOUNT_CREATED: t => t('account:account-created', 'Account created'),
    SIGN_IN: t => t('account:sign-in', 'Sign In')
};

const startAccMgEvent = new CustomEvent(events.START_ACC_MANAGEMENT);
const exitAccMgEvent = new CustomEvent(events.EXIT_ACC_MANAGEMENT);

const useNavigationState = (props = { view: 'MENU' }) => {
    const [currentView, setCurrentView] = useState(props.view);
    const [t] = useTranslation('account');

    const {
        mountingPoints: { navPanel }
    } = useConfigContext();

    const navigationPanel = document.querySelector(navPanel);

    const triggerNavigationEvent = title => {
        const event = new CustomEvent('aem.accmg.step', { detail: { title } });
        triggerEvent(event);
    };

    const triggerEvent = event => navigationPanel && navigationPanel.dispatchEvent(event);

    const handleBack = useCallback(() => {
        if (currentView === null) {
            return;
        }
        const parent = ancestors[currentView];
        if (parent === 'MENU') {
            // no parent view means we're out of the account management process and back to navigation
            // so we're resetting the title
            showMenu();
            return;
        }
        if (parent === 'MY_ACCOUNT') {
            showMyAccount();
            return;
        }
        if (parent) {
            switchTo(parent);
        }
    }, [currentView]);

    useEventListener(document, 'aem.navigation.back', handleBack);

    const switchTo = view => {
        triggerNavigationEvent(stepTitles[view](t));
        setCurrentView(view);
    };

    const showSignIn = () => {
        triggerEvent(startAccMgEvent);
        switchTo('SIGN_IN');
    };
    const showMenu = () => {
        triggerEvent(exitAccMgEvent);
        setCurrentView('MENU');
    };
    const showMyAccount = () => {
        dispatchEvent(startAccMgEvent);
        switchTo('MY_ACCOUNT');
    };
    const showChangePassword = () => {
        switchTo('CHANGE_PASSWORD');
    };
    const showCreateAccount = () => {
        switchTo('CREATE_ACCOUNT');
    };
    const showForgotPassword = () => {
        switchTo('FORGOT_PASSWORD');
    };
    const showAccountCreated = () => {
        switchTo('ACCOUNT_CREATED');
    };

    const api = {
        showSignIn,
        showMenu,
        showMyAccount,
        showChangePassword,
        showForgotPassword,
        showCreateAccount,
        showAccountCreated
    };

    return [currentView, { ...api, switchTo, handleBack }];
};

export default useNavigationState;
