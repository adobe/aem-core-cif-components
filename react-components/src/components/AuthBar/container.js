/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';

import AuthBar from './authBar';
import AuthModal from '../AuthModal';
import { useNavigationContext } from '../../context/NavigationContext';

import classes from './container.css';

const Container = () => {
    const [{ view }] = useNavigationContext();

    const hasModal = view !== 'MENU';
    const modalClassName = hasModal ? classes.modal_open : classes.modal;

    return (
        <>
            <div className="navigation__footer">
                <AuthBar />
            </div>
            {view !== null && (
                <div className={modalClassName}>
                    <AuthModal />
                </div>
            )}
        </>
    );
};

export default Container;
