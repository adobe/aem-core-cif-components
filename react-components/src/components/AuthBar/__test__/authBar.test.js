/*******************************************************************************
 *
 *    Copyright 2021 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import { render } from 'test-utils';

const { TextEncoder } = require('util');
const { Crypto } = require('@peculiar/webcrypto');

import AuthBar from '../authBar';
import UserContextProvider from '../../../context/UserContext';

describe('<AuthBar>', () => {
    beforeAll(() => {
        window.TextEncoder = TextEncoder;
        window.crypto = new Crypto();

        window.document.body.setAttributeNode(document.createAttribute('data-cmp-data-layer-enabled'));

        window.adobeDataLayer = [];
        window.adobeDataLayer.getState = jest.fn(ref => {
            return { result: ref };
        });
        window.adobeDataLayer.push = jest.fn();
        window.adobeDataLayer.addEventListener = jest.fn();
        window.adobeDataLayer.removeEventListener = jest.fn();
    });

    beforeEach(() => {
        window.adobeDataLayer.getState.mockClear();
        window.adobeDataLayer.push.mockClear();
        window.adobeDataLayer.addEventListener.mockClear();
        window.adobeDataLayer.removeEventListener.mockClear();
    });

    it('renders the component for anonymous user', () => {
        const handler = jest.fn(state => state);

        const { asFragment } = render(
            <UserContextProvider reducerFactory={() => handler}>
                <AuthBar />
            </UserContextProvider>
        );

        expect(asFragment()).toMatchSnapshot();
        expect(window.adobeDataLayer.push).toHaveBeenCalledTimes(1);
        expect(window.adobeDataLayer.push).toHaveBeenCalledWith({ user: null });
    });

    it('renders the component for logged user', () => {
        const handler = jest.fn(state => state);

        const initialState = {
            currentUser: {
                firstname: 'Test',
                lastname: 'User',
                email: 'test_user@example.com',
                addresses: []
            },
            isSignedIn: true
        };

        const { asFragment } = render(
            <UserContextProvider initialState={initialState} reducerFactory={() => handler}>
                <AuthBar />
            </UserContextProvider>
        );
        expect(asFragment()).toMatchSnapshot();
        expect(window.adobeDataLayer.push).toHaveBeenCalledTimes(1);
        expect(window.adobeDataLayer.push).toHaveBeenCalledWith({
            user: {
                firstname: 'Test',
                lastname: 'User',
                email: 'test_user@example.com',
                addresses: []
            }
        });
    });
});
