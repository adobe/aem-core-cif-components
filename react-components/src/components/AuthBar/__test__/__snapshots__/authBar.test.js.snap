// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthBar> renders the component for anonymous user 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <button
      class="root_highPriority"
      type="button"
    >
      <span
        class="content"
      >
        Sign In
      </span>
    </button>
  </div>
</DocumentFragment>
`;

exports[`<AuthBar> renders the component for logged user 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <button
      class="root"
    >
      <span
        class="content"
      >
        <span
          class="avatar"
        >
          <span
            class="root"
          >
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
              />
              <circle
                cx="12"
                cy="7"
                r="4"
              />
            </svg>
          </span>
        </span>
        <span
          class="user"
        >
          <span
            class="fullName"
          >
            Test User
          </span>
          <span
            class="email"
          >
            test_user@example.com
          </span>
        </span>
        <span
          class="icon"
        >
          <span
            class="root"
          >
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="9 18 15 12 9 6"
              />
            </svg>
          </span>
        </span>
      </span>
    </button>
  </div>
</DocumentFragment>
`;
